src/components/categoricaldisplay/ResultsPanel.tsx

```diff
@@ -111,6 +113,61 @@ const CategoryTitle = ({ category, activeHeadKey, isNestedList, handleTitleClick
     </div>
 );
 
+const ROWsubCategoryMapper: {[key: string]: string} = {
+    commodity_basket: 'basket-indices/?tab=options#commodities-basket',
+    forex_basket: 'basket-indices/?tab=options#forex-basket',
+    random_index: 'synthetic/?tab=options#continuous-indices',
+    crash_index: 'synthetic/?tab=multipliers#crash-boom',
+    random_daily: 'synthetic/?tab=options#daily-reset-indices',
+    jump_index: 'synthetic/?tab=options#jump-indices',
+    step_index: 'synthetic/?tab=multipliers#step-indices',
+    major_pairs: 'forex/?tab=options#major-pairs',
+    minor_pairs: 'forex/?tab=options#minor-pairs',
+    americas_OTC: 'stock/?tab=options#american-indices',
+    asia_oceania_OTC: 'stock/?tab=options#asian-indices',
+    europe_OTC: 'stock/?tab=options#european-indices',
+    non_stable_coin: 'cryptocurrencies/?tab=multipliers#crypto-pairs',
+    energy: 'commodities/?tab=options#energy',
+    metals: 'commodities/?tab=options#metals',
+};
+
+const EUsubCategoryMapper: {[key: string]: string} = {
+    random_index: 'synthetic/?tab=multipliers#continuous-indices',
+    crash_index: 'synthetic/?tab=multipliers#crash-boom',
+    major_pairs: 'forex/?tab=multipliers#major-pairs',
+    non_stable_coin: 'cryptocurrencies/?tab=multipliers#crypto-pairs',
+};
+
+const redirectLink = (subCategoryId: string, is_eu_client: boolean) => {
+    const DEFAULT_LANGUAGE = 'EN';
+    const lang_from_url = new URLSearchParams(window.location.search).get('lang')?.toLowerCase() || DEFAULT_LANGUAGE.toLowerCase();
+    const add_EU_suffix = is_eu_client ? 'eu.' : '';
+    const link_mapper = is_eu_client ? EUsubCategoryMapper[subCategoryId] : ROWsubCategoryMapper[subCategoryId];
+    let link;
+    if (!link_mapper) {
+        if(is_eu_client) {
+            link = `https://eu.deriv.com/${lang_from_url}`;
+        }else {
+            link = `https://deriv.com/${lang_from_url}`;
+        }
```

Comment:
```suggestion
        link = `https://${is_eu_client ? 'eu' : ''}.deriv.com/${lang_from_url}`;
```
@henry-deriv please also run option+shift_f in the changed files. they are not prettified